src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -352,22 +352,49 @@ export class StudyViewPageStore {
     @action
     updateNAValue = (uniqueKey: string): void => {
         let newValue = this.isShowNAChecked(uniqueKey);
-        let newFilter = _.clone(this._clinicalDataBinFilterSet.get(uniqueKey))!;
-        newFilter.showNA = newValue;
-        this._clinicalDataBinFilterSet.set(uniqueKey, newFilter);
+        if (this.isGenericAssayChart(uniqueKey)) {
+            let newFilter = _.clone(
+                this._genericAssayDataBinFilterSet.get(uniqueKey)
+            )!;
+            newFilter.showNA = newValue;
+            this._genericAssayDataBinFilterSet.set(uniqueKey, newFilter);
+        } else {
+            let newFilter = _.clone(
+                this._clinicalDataBinFilterSet.get(uniqueKey)
+            )!;
+            newFilter.showNA = newValue;
+            this._clinicalDataBinFilterSet.set(uniqueKey, newFilter);
+        }
     };
 
     @action
     toggleNAValue = (uniqueKey: string): void => {
         this.updateNAValue(uniqueKey);
-        let newFilter = _.clone(this._clinicalDataBinFilterSet.get(uniqueKey))!;
-        let showNA = newFilter.showNA;
-        newFilter.showNA = !showNA;
-        this._clinicalDataBinFilterSet.set(uniqueKey, newFilter);
+        if (this.isGenericAssayChart(uniqueKey)) {
```

Comment:
seems this code is quite redundant.   how about:
```
targetWhateverSet = this.isGenericAssChart(uniqueKey);
// now proceed with cloning and setting
```

Reply:
I tried this but I get an error when trying to set newFilter to targetWhateverSet (targetWhateverSet.set(uniqueKey, newFilter)) as newFilter can have two types GenericAssayDataBinFilter or ClinicalDataBinFilter.
However, I modified a bit the method to be less redundant.